### DISCUSSION TITLE
Feature Request: Filter tests in a certain group

### DISCUSSION THREAD

**Discussion by tisonkun** [nextest-rs/nextest#1756] [Feature requests]
Nextest currently suppose grouping tests to run exclusively.

However, the filter DSL works in a irrelevant way, so that when I'd like to run a specific test group, I have to write the filter DSL twice and ensure they are the same.

It is possible to have a filter DSL such as `group(group_name)` to match all the tests in a certain group?

Maybe related to https://github.com/nextest-rs/nextest/issues/101 but subtly different.

**Comment by sunshowers** [comment]
Thanks! Good idea and one I'd been thinking about.

One issue is that this can introduce cycles in the execution graph for the DSL interpreter, but that's not too hard to handle. Will try getting to it in the next few weeks.

**Reply by JP-Ellis** [reply]
I was curious whether this was still on the roadmap? From what I can tell, there's no `group` selector yet in the filterset DSL.

**Reply by sunshowers** [reply]
Yes, it is -- it needs someone to step up and implement the feature, keeping things like cycles in mind.

**Comment by sunshowers** [comment]
See https://github.com/nextest-rs/nextest/pull/3273.

**Comment by sunshowers** [comment]
This is now out in nextest 0.9.133.
